gdx/src/com/badlogic/gdx/graphics/g3d/utils/Debug3dRenderer.java

```diff
@@ -0,0 +1,119 @@
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.g3d.utils;
+
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Mesh;
+import com.badlogic.gdx.graphics.VertexAttribute;
+import com.badlogic.gdx.graphics.VertexAttributes;
+import com.badlogic.gdx.graphics.VertexAttributes.Usage;
+import com.badlogic.gdx.graphics.g3d.Material;
+import com.badlogic.gdx.graphics.g3d.Renderable;
+import com.badlogic.gdx.graphics.g3d.RenderableProvider;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Disposable;
+import com.badlogic.gdx.utils.GdxRuntimeException;
+import com.badlogic.gdx.utils.Pool;
+
+/** Debug3dRenderer allows to see box, camera in 3d space.
+ * <p>
+ * How to use it :
+ * </p>
+ * 
+ * <pre>
+ * // Create debugger
+ * Debug3dRenderer debugger = new Debug3dRenderer();
+ * // Each Frame
+ * MeshPartBuilder builder = debugger.begin();
+ * FrustumShapeBuilder.build(builder, camera);
+ * BoxShapeBuilder.build(builder, box);
+ * debugger.end()
+ * // Render
+ * modelBatch.render(debugger);
+ * // After using it
+ * debugger.dispose();
+ * </pre>
+ * 
+ * @author realitix */
+public class Debug3dRenderer implements Disposable, RenderableProvider {
+
+	/** Final renderable */
+	private final Renderable renderable = new Renderable();
+
+	/** Builder used to update the mesh */
+	private final MeshBuilder builder;
+
+	/** Mesh being rendered */
+	private final Mesh mesh;
+
+	private boolean building;
+
+	/** Create a Debug3dRenderer */
+	public Debug3dRenderer () {
+		this(5000, 5000);
+	}
+
+	/** Create a Debug3dRenderer with parameters
+	 * @param maxVertices max vertices in mesh
+	 * @param maxIndices max indices in mesh */
+	public Debug3dRenderer (int maxVertices, int maxIndices) {
+		VertexAttributes attributes = new VertexAttributes(new VertexAttribute(Usage.Position, 3, "a_position"),
+			new VertexAttribute(Usage.ColorPacked, 4, "a_color"));
+
+		// Init mesh
+		mesh = new Mesh(false, maxVertices, maxIndices, attributes);
+
+		// Init builder
+		builder = new MeshBuilder();
+
+		// Init renderable
+		renderable.meshPart.mesh = mesh;
+		renderable.meshPart.primitiveType = GL20.GL_LINES;
+		renderable.material = new Material();
+	}
+
+	/** Initialize Debug3dRender for mesh generation */
+	public MeshPartBuilder begin () {
+		if (building) throw new GdxRuntimeException("Call end() after calling begin()");
+		building = true;
+
+		builder.begin(mesh.getVertexAttributes(), GL20.GL_LINES);
+		return builder;
```

Comment:
Better change this in `builder.begin(attributes);` followed by `builder.part(id, primitive, renderable.meshPart);`. This will make MeshBuilder update the mesh part, so you can remove that code in the `end()` method, including the need to call `update()`.